server/homebrew.api.js

```diff
@@ -89,148 +129,177 @@ const newBrew = async (req, res)=>{
 
 	beforeNewSave(req.account, brew);
 
-	let saved;
+	const newHomebrew = new HomebrewModel(brew);
+	newHomebrew.editId = nanoid(12);
+	newHomebrew.shareId = nanoid(12);
+
+	let googleId, saved;
 	if(transferToGoogle) {
-		saved = await newGoogleBrew(req.account, brew, res)
+		googleId = await newGoogleBrew(req.account, newHomebrew, res)
 			.catch((err)=>{
-				res.status(err.status || err.response.status).send(err.message || err);
+				res.status(err?.status || err?.response?.status || 500).send(err?.message || err);
 			});
+		excludeStubProps(newHomebrew);
 	} else {
-		saved = await newLocalBrew(brew)
-			.catch((err)=>{
-				res.status(500).send(err);
-			});
+		// Compress brew text to binary before saving
+		newHomebrew.textBin = zlib.deflateRawSync(newHomebrew.text);
+		// Delete the non-binary text field since it's not needed anymore
+		newHomebrew.text = undefined;
+	}
+	if(transferToGoogle && !googleId) {
+		if(!res.headersSent) {
+			res.status(500).send('Unable to save document to Google Drive');
+		}
+		return;
+	} else if(transferToGoogle) {
+		newHomebrew.googleId = googleId;
 	}
+
+	saved = await newHomebrew.save()
+		.catch((err)=>{
+			console.error(err, err.toString(), err.stack);
+			throw `Error while creating new brew, ${err.toString()}`;
+		});
 	if(!saved) return;
+	saved = saved.toObject();
+
 	return res.status(200).send(saved);
 };
```

Comment:
It's not letting me do a suggested edit, but moving the extra `if(transferToGoogle)` checks up:

```
	let googleId, saved;
	if(transferToGoogle) {
	//Save brew to Google Drive; newHomebrew will just be a stub
		googleId = await newGoogleBrew(req.account, newHomebrew, res)
			.catch((err)=>{
				res.status(err?.status || err?.response?.status || 500).send(err?.message || err);
			});
		if(!googleId) {
			res.status(500).send('Unable to save document to Google Drive');
			return;
		}
		newHomebrew.googleId = googleId;
		excludeStubProps(newHomebrew);
	} else {
	//Prep brew for Mongo; newHomebrew will be the full brew
		// Compress brew text to binary before saving
		newHomebrew.textBin = zlib.deflateRawSync(newHomebrew.text);
		// Delete the non-binary text field since it's not needed anymore
		newHomebrew.text = undefined;
	}

	saved = await newHomebrew.save()
		.catch((err)=>{
			console.error(err, err.toString(), err.stack);
			throw `Error while creating new brew, ${err.toString()}`;
		});
	if(!saved) return;
	saved = saved.toObject();
	return res.status(200).send(saved);
};
```